src/screen/GameScreen2P.java

```diff
@@ -0,0 +1,350 @@
+package screen;
+
+import engine.Cooldown;
+import engine.Core;
+import engine.GameSettings;
+import engine.GameState;
+import entity.*;
+
+import java.awt.event.KeyEvent;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.Callable;
+
+/**
+ * Implements the game screen, where the action happens.
+ *
+ * @author <a href="mailto:RobertoIA1987@gmail.com">Roberto Izquierdo Amo</a>
+ *
+ */
+public class GameScreen2P extends Screen implements Callable<Integer> {
```

Comment:
I think, you can just extends `GameScreen` and override several methods. Please try my opinion